cou/utils/openstack.py

```diff
@@ -464,3 +409,48 @@ def is_charm_supported(charm: str) -> bool:
     :rtype: bool
     """
     return bool(OpenStackCodenameLookup.lookup(charm)) or charm in SUBORDINATES
+
+
+def _generate_track_mapping() -> dict[TrackKeys, str]:
+    """Generate the track mapping for the auxiliary charms.
+
+    This mapping should be updated periodically by adding new lines in the file
+    openstack_to_track_mapping.csv
+
+    See the following url for more details:
+    https://docs.openstack.org/charm-guide/latest/project/charm-delivery.html
+
+    :return: Dictionary containing the tracks by charm name, series and OpenStack release.
+    :rtype: dict[namedtuple, str]
+    """
+    track_mapping = {}
+    with open(
+        Path(__file__).parent / "openstack_to_track_mapping.csv",
+        encoding=encodings.utf_8.getregentry().name,
+    ) as csv_file:
+        csv_reader = csv.DictReader(csv_file, delimiter=",")
+        for row in csv_reader:
+            track_key = TrackKeys(
+                charm=row["charm"], series=row["series"], os_release=row["os_release"]
+            )
+            track_mapping[track_key] = row["track"]
+    return track_mapping
+
+
+OPENSTACK_TO_TRACK_MAPPING = _generate_track_mapping()
+
+
+def openstack_to_track(charm: str, series: str, os_release: OpenStackRelease) -> Optional[str]:
```

Comment:
not-blocking: I think that this function is unnecessary. Since you can use
```python
>>> OPENSTACK_TO_TRACK_MAPPING[("ceph-mon", "focal", "ussuri")]
'octopus'
>>> OPENSTACK_TO_TRACK_MAPPING.get(("ceph-mon", "focal", "ussuri"))
'octopus'
```

Reply:
True, but it provides a nicer interface for callers that are outside this module. If you remove it, the caller needs to import `OPENSTACK_TO_TRACK_MAPPING` and understand its internal structure.

Consider also that pretty early on when we were discussing using or not using zaza utils, we decided to try to keep the "openstack utilities" as separate as possible from the rest of COU, so that we would at some point be able to release them as a separate package (whether that will actually ever happen is a separate question)

Reply:
I ended up removing this function :grimacing:. I've added a reverse map to be able to know the compatible OpenStack releases of a certain track. Let me know if I should re-add this function

Reply:
I see point of easy to use, but i do not think that COU will be used as package, so that's I still think it's not important. For developer is good to be able to understand the structure and still I will check what function is returning to understand it. As developer that will be good chance that I use auto import and that will import `OPENSTACK_TO_TRACK_MAPPING` instead of this function. 

Now I read second paragraph and see that we want to use openstack utils as potential package. I do not have strong opinion here, so I think it's up to you @gabrielcocenza.

Reply:
After checking again ho function is used, maybe we can pack more to this function e.g. 
```python
def get_track(charm: str, series: str, os_release: OpenStackRelease) -> Optional[str]:
    track = OPENSTACK_TO_TRACK_MAPPING.get(self.charm, self.series, target)
    if track:
        return f"{track}/stable"
    
    return track
```
That will make more sense to me than packed basically `OPENSTACK_TO_TRACK_MAPPING.get(self.charm, self.series, target)` in function.